### ISSUE TITLE
Abort trap: 6

### ISSUE THREAD

**Issue by aubergene** [plutomonkey/zcash-vanity#1]
It exits after a few seconds with this error. Additionally it doesn't seem to respond to ctrl-c to exit cleanly.

```
zcash-vanity -ci -- zcauber
Available devices on platform Apple:
  0:0 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
  0:1 Iris
zcAuBErGoVrpxMJ1kAuj21BEU6EotDBdc4VKCx5NJ5a7XTVMy5KAZ6jotjhPCirHWuxRRp8KNoWeeg5sPE2ZPrm1JbTMYoj
SKxsPsozZ3b23HkdcuD3gGLuYBDJ7VrnMGNcqCthvt6Zkv3VG7cj
zcAubeRWuyfXFfBREDa1tgHwfTqNjs8E8Pn34tZX4N57KzRD1HJPJb1YARhm9nn5TQHK4dqoNowSd8TiQQjGGtDNZ6gWjBp
SKxrTZ7saFLhwvDa7XHK8r1A9c9wyLou2Y2XTiFVaKFSXfPAS5MZ
zcauber89brTJnpVUYxzyiZkajNsJYKWMwHcwaxNsGeTdqnpXPNMw8FtNFERPRnUE6EyLJfKzjNWpsx9VeK7LRuTEt336r8
SKxtnn9gnQWoFMvYAnyKe7xPxvnX29Jkvn2VrSgFeYUECD3VADXe
zcAuBerUC1BBcdGnxLDBdGnugavRjpHQLkXu4ipScWEz9yBQXJRGnM7hSDyCDTvYP5D6jkEcsFGVcxm4tNTGSRVNYMPKMyD
SKxoZFgHStASc3nErb9yNPhnj9CsfmyVh1NmHvZqDJcaWQ7F3ujx
zcauBeRBkzgN2tPVqzf7tFVvg22QhUXhyb6ap6T8Ssw3n6WGeLKJymCfPUgwCpZveMyv6EUu2kPrvnWVBhwWnsD2iqQFuei
SKxuo58GwoGRcNPuXDWmSP1AVuRMyWomY2fLWpiPt4gQJugHfJv5
zcaUbeRmrmGWyZ5CkwvJdp5CmAKS9tkkcJRq2ocNXTmfY1PhSDm8gb7QLAaYLM3yFaJZLj3Q91VsCq8kdYDG9H1AHL8B11F
SKxtwe1Wdkw7i1e2aB4xnDH1CeYhh6zgPFC348u5zKxDy9B34RGn
Elapsed: 6/3s Rate: 3702784/s Prob: 84.74%Abort trap: 6
```

**Comment by jasondavies** [comment]
Thanks for the report.  Does it exhibit the same behaviour if you limit to a single device, e.g. `-d 0:0` or `-d 0:1`?

I'm working on a new version using Vulkan (or Metal for Apple), but will try and fix any problems in the current version if we can get to the bottom of them.

**Comment by jasondavies** [comment]
Just published version 0.2.4, which simplifies data handling (a possible suspect for this issue).  Let me know if that helps.

**Comment by aubergene** [comment]
I updated, but still getting the same issue. Let me know if anything I can try to help debug. 

Also, it still reports as v 0.0.1 from --help [from here](https://github.com/plutomonkey/zcash-vanity/blob/master/src/main.rs#L33)
